api/tests/App/Entity/Repository/OrganisationRepositoryTest.php

```diff
@@ -0,0 +1,152 @@
+        self::assertCount(4, $nonDeletedOrgs);
+    }
+
+    /** @test */
+    public function testHasActiveEntities()
+    {
+        $orgs = $this->fixtures->createOrganisations(3);
+        $user = $this->fixtures->createUser()->setRoleName(\App\Entity\User::ROLE_PA);
+        $clientActive = $this->fixtures->createClient($user);
+        $clientArchived = $this->fixtures->createClient($user, ['setArchivedAt' => new \DateTime()]);
+        $clientArchived2 = $this->fixtures->createClient($user, ['setArchivedAt' => new \DateTime()]);
+        $clientDeleted = $this->fixtures->createClient($user, ['setDeletedAt' => new \DateTime()]);
+        $clientDeleted2 = $this->fixtures->createClient($user, ['setDeletedAt' => new \DateTime()]);
+
+
+        $this->em->flush();
+
+        //Org1 with no entities should have no active entities.
+        $result = $this->sut->hasActiveEntities($orgs[0]->getId());
+        self::assertFalse($result);
+
+        //Org1 with 1 soft deleted client should have no active entities.
+        $this->fixtures->addClientToOrganisation($clientDeleted->getId(), $orgs[0]->getId());
+        $this->em->flush();
+        $result = $this->sut->hasActiveEntities($orgs[0]->getId());
+        self::assertFalse($result);
+
+        //Org1 with 1 soft deleted client
+        //And 1 archived client
+        //Should have no active entities.
+        $this->fixtures->addClientToOrganisation($clientArchived->getId(), $orgs[0]->getId());
+        $this->em->flush();
+        $result = $this->sut->hasActiveEntities($orgs[0]->getId());
+        self::assertFalse($result);
+
+        //Org1 with 1 soft deleted client
+        //And 1 archived client
+        //And 1 active client
+        //Should have active entities.
+        $this->fixtures->addClientToOrganisation($clientActive->getId(), $orgs[0]->getId());
+        $this->em->flush();
+        $result = $this->sut->hasActiveEntities($orgs[0]->getId());
+        self::assertTrue($result);
+
+        //Org2 with 1 user should have active entities.
+        $this->fixtures->addUserToOrganisation($user->getId(), $orgs[1]->getId());
+        $this->em->flush();
+        $result = $this->sut->hasActiveEntities($orgs[1]->getId());
+        self::assertTrue($result);
+
+        //Org2 with 1 user
+        //And 1 soft deleted client
+        //should have active entities.
+        $this->fixtures->addClientToOrganisation($clientDeleted2->getId(), $orgs[1]->getId());
+        $this->em->flush();
+        $result = $this->sut->hasActiveEntities($orgs[1]->getId());
+        self::assertTrue($result);
+
+        //Org2 with 1 user
+        //And 1 soft deleted client
+        //And 1 archived client
+        //should have active entities.
+        $this->fixtures->addClientToOrganisation($clientArchived2->getId(), $orgs[1]->getId());
+        $this->em->flush();
+        $result = $this->sut->hasActiveEntities($orgs[1]->getId());
+        self::assertTrue($result);
```

Comment:
The tests all look great but it may be easier to read if the different cases are split out into their own functions:

```suggestion
/** @test */
    public function testHasActiveEntities_no_entities_in_org_returns_false()
    {
         ...
     }
     
     /** @test */
    public function testHasActiveEntities_soft_deleted_client_in_org_returns_false()
    {
         ...
     }
     
     etc
```

This way when tests fail in the future you know exactly what's gone wrong rather than having to work out which scenario in a big bunch of assertions have failed.